pkg/daemon/ceph/multus/network_types.go

```diff
@@ -0,0 +1,110 @@
+	// of executing code in an incorrect namespace will be greater.  See
+	// https://github.com/golang/go/wiki/LockOSThread for further details.
+	//
+	// NOTE: modified to make the toRun func take no argument since we don't use it and since it
+	// makes it harder to fit the linux wrapper onto the type.
+	Do(toRun func() error) error
+
+	// Returns the filesystem path representing this object's network namespace
+	Path() string
+
+	// Returns a file descriptor representing this object's network namespace
+	Fd() uintptr
+}
+
+// Create an executor that will allow us to mock all the commands we make for unit testing
+type NetworkExecutor interface {
+	// from "github.com/containernetworking/plugins/pkg/ns"
+	// https://pkg.go.dev/github.com/containernetworking/plugins/pkg/ns#GetCurrentNS
+	//
+	// NOTE: when this is called, the NetNS object returned will have a Path() method that does not
+	// return consistent results between different calls to GetCurrentNS(). This is because the
+	// function gets the /proc path to whatever OS thread the goroutine is currently running on
+	// which may change.
+	GetCurrentNS() (NetNS, error)
+
+	// from "github.com/containernetworking/plugins/pkg/ns"
+	// https://pkg.go.dev/github.com/containernetworking/plugins/pkg/ns#GetNS
+	GetNS(nspath string) (NetNS, error)
+
+	// from "net"
+	// https://pkg.go.dev/net?utm_source=gopls#Interfaces
+	Interfaces() ([]net.Interface, error)
+
+	// from "github.com/vishvananda/netlink"
+	// https://pkg.go.dev/github.com/vishvananda/netlink#LinkByName
+	LinkByName(name string) (netlink.Link, error)
+
+	// from "github.com/vishvananda/netlink"
+	// https://pkg.go.dev/github.com/vishvananda/netlink#LinkSetDown
+	LinkSetDown(link netlink.Link) error
+
+	// from "github.com/vishvananda/netlink"
+	// https://pkg.go.dev/github.com/vishvananda/netlink#LinkSetUp
+	LinkSetUp(link netlink.Link) error
+
+	// from "github.com/vishvananda/netlink"
+	// https://pkg.go.dev/github.com/vishvananda/netlink#LinkAdd
+	LinkAdd(link netlink.Link) error
+
+	// from "github.com/vishvananda/netlink"
+	// https://pkg.go.dev/github.com/vishvananda/netlink#LinkDel
+	LinkDel(link netlink.Link) error
+
+	// from "github.com/vishvananda/netlink"
+	// https://pkg.go.dev/github.com/vishvananda/netlink#LinkSetNsFd
+	LinkSetNsFd(link netlink.Link, fd int) error
+
+	// from "github.com/vishvananda/netlink"
+	// https://pkg.go.dev/github.com/vishvananda/netlink#AddrList
+	AddrList(link netlink.Link, family int) ([]netlink.Addr, error)
+
+	// from "github.com/vishvananda/netlink"
+	// https://pkg.go.dev/github.com/vishvananda/netlink#AddrAdd
+	AddrAdd(link netlink.Link, addr *netlink.Addr) error
+
+	// from "github.com/vishvananda/netlink"
+	// https://pkg.go.dev/github.com/vishvananda/netlink#RouteList
+	RouteList(link netlink.Link, family int) ([]netlink.Route, error)
+
+	// from "github.com/vishvananda/netlink"
+	// https://pkg.go.dev/github.com/vishvananda/netlink#RouteAdd
+	RouteAdd(route *netlink.Route) error
+}
```

Comment:
Do we need/use this anywhere?

Reply:
Currently, I only unit test one of the network functions. But this allows expanding the unit tests without writing more of the boilerplate.